pkg/extender/extender.go

```diff
@@ -357,10 +363,10 @@ func (e *Extender) processPrioritizeRequest(w http.ResponseWriter, req *http.Req
 			zoneInfo.PreferredLocality = zoneInfo.PreferredLocality[:0]
 			regionInfo.PreferredLocality = regionInfo.PreferredLocality[:0]
 			for _, node := range volume.DataNodes {
-				log.Debugf("ID: %v Hostname: %v", node, idMap[node])
-				regionInfo.PreferredLocality = append(regionInfo.PreferredLocality, regionInfo.HostnameMap[idMap[node]])
-				zoneInfo.PreferredLocality = append(zoneInfo.PreferredLocality, zoneInfo.HostnameMap[idMap[node]])
-				rackInfo.PreferredLocality = append(rackInfo.PreferredLocality, rackInfo.HostnameMap[idMap[node]])
+				log.Debugf("ID: %v Hostname: %v", node, idMap[node].Hostname)
```

Comment:
If `Driver.GetPodVolumes()` returns volumes which have an outdated data node which is no longer present in Driver.GetNodes(), then we could panic here since `node` won't be present in `idMap`.

Reply:
Yes, can happen in the rare case if nodes are decommissioned right between those 2 calls. Updated to ignore missing nodes.